bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/groovy/org/eclipse/smarthome/core/thing/xml/test/SystemWideChannelTypesTest.groovy

```diff
@@ -0,0 +1,128 @@
+        assertThat bundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes + 1)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes + 1)
+
+        // uninstall test bundle
+        bundle.uninstall();
+        assertThat bundle.state, is(Bundle.UNINSTALLED)
+
+        thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes)
+
+    }
+
+    @Test
+    void 'assert that System Channels are used by other binding'() {
+        def bundleContext = getBundleContext()
+        def initialNumberOfThingTypes = thingTypeProvider.getThingTypes(null).size()
+        def initialNumberOfChannelTypes = systemChannelTypeProvider.getSystemChannelTypes().size()
+
+
+        // install test bundle
+        Bundle sysBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_BUNDLE_NAME)
+        assertThat sysBundle, is(notNullValue())
+
+        Bundle sysUserBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_USER_BUNDLE_NAME)
+        assertThat sysUserBundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes + 2)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes + 1)
+
+    }
+
+    @Test
+    void 'assert that Thing Types have proper channel definitions'() {
+        def bundleContext = getBundleContext()
+        def initialNumberOfThingTypes = thingTypeProvider.getThingTypes(null).size()
+        def initialNumberOfChannelTypes = systemChannelTypeProvider.getSystemChannelTypes().size()
+
+
+        // install test bundle
+        Bundle sysBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_BUNDLE_NAME)
+        assertThat sysBundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null).findAll{it.getUID().getId().equals("wireless-router") }
+
+        for(ThingType type : thingTypes) {
+
+            assertThat type.getChannelDefinitions().size(),is(3)
+            int i = 3;
+            for(ChannelDefinition deff : type.getChannelDefinitions()) {
+                if((deff.getId().equals("test") && deff.getType().UID.getAsString().equals("system:my-channel"))
+                || (deff.getId().equals("sigstr") && deff.getType().UID.getAsString().equals("system:signal-strength"))
+                || (deff.getId().equals("lowbat") && deff.getType().UID.getAsString().equals("system:low-battery"))) {
+                    --i;
+                }
+            }
+
+            assertThat i, is(0)
```

Comment:
At first sight it is not clear what is really checked here. Taking also the previous comment in consideration please rewrite this to search for the channel types and check if there are present and also contain the defined properties.